src/clj/puppetlabs/services/jruby_pool_manager/impl/jruby_internal.clj

```diff
@@ -232,14 +244,8 @@
                           "maximum number of borrows (%s)")
                      (:id instance)
                      max-borrows)
-          (try
-            (flush-instance-fn pool instance)
-            (finally
-              (.releaseItem pool instance false))))
-        (.releaseItem pool instance)))
-    ;; if we get here, it was from a borrow and we got a Retry, so we just
-    ;; return it to the pool.
-    (.releaseItem (:pool instance) instance)))
+          (flush-instance-fn instance))
+        (.releaseItem pool instance)))))
```

Comment:
So before, that `(finally (.releaseItem pool instance false))` block was a noop. Passing in `false` to `.releaseItem` became a noop at some point in the past I think. @camlow325 might know why. But we decided to remove that overload of `.releaseItem`

The `flush-instance-fn` (used to and still does) unregisters the old instance, creates the new instance, and adds it to the pool